comms/dht/src/crypt.rs

```diff
@@ -325,4 +379,163 @@ mod test {
+            pad_message[prepend_message.len()..prepend_message.len() + message.len()]
+        );
+        // pad is well specified
+        assert_eq!(pad, pad_message[prepend_message.len() + message.len()..]);
+
+        // test for empty message
+        let message: [u8; 0] = [];
+        let prepend_message = message.len().to_le_bytes();
+        let pad_message = pad_message_to_base_length_multiple(&message);
+        let pad = [0u8; 120];
+
+        // padded message is of correct length
+        assert_eq!(pad_message.len(), MESSAGE_BASE_LENGTH);
+        // prepend message is well specified
+        assert_eq!(prepend_message, pad_message[..prepend_message.len()]);
+        // message body is well specified
+        assert_eq!(
+            message,
+            pad_message[prepend_message.len()..prepend_message.len() + message.len()]
+        );
+
+        // pad is well specified
+        assert_eq!(pad, pad_message[prepend_message.len() + message.len()..]);
+    }
+
+    #[test]
+    fn get_original_message_from_padded_text_successful() {
+        // test for short message
+        let message = vec![0u8, 10, 22, 11, 38, 74, 59, 91, 73, 82, 75, 23, 59];
+        let pad_message = pad_message_to_base_length_multiple(message.as_slice());
+
+        let output_message = get_original_message_from_padded_text(pad_message.as_slice()).unwrap();
+        assert_eq!(message, output_message);
+
+        // test for large message
+        let message = vec![100u8; 1024];
+        let pad_message = pad_message_to_base_length_multiple(message.as_slice());
+
+        let output_message = get_original_message_from_padded_text(pad_message.as_slice()).unwrap();
+        assert_eq!(message, output_message);
+
+        // test for base message of base length
+        let message = vec![100u8; 984];
+        let pad_message = pad_message_to_base_length_multiple(message.as_slice());
+
+        let output_message = get_original_message_from_padded_text(pad_message.as_slice()).unwrap();
+        assert_eq!(message, output_message);
+
+        // test for empty message
+        let message: Vec<u8> = vec![];
+        let pad_message = pad_message_to_base_length_multiple(message.as_slice());
+
+        let output_message = get_original_message_from_padded_text(pad_message.as_slice()).unwrap();
+        assert_eq!(message, output_message);
+    }
+
+    #[test]
+    fn decryption_fails_if_pad_message_prepend_is_modified() {
+        let pk = CommsPublicKey::default();
+        let key = CipherKey(*chacha20::Key::from_slice(pk.as_bytes()));
+        // long text makes last test case deterministic
+        let message = "This is my secret message, keep it secret !".as_bytes().to_vec();
+        let mut encrypted = encrypt(&key, &message);
+
+        // failure in case message length prepending has been modified such that resulting
+        // length is too big to fit within pad message length
+        encrypted[size_of::<Nonce>() + 1] += 1;
```

Comment:
It's possible for the value at `encrypted[size_of::<Nonce>() + 1]` to be 255 - 255 + 1 will overflow so the test will be flaky.

Also increasing the ciphertext byte won't necessarily result in an increase in the plaintext byte. This is observed when `unwrap_err` panics sometimes. 

Suggest rather testing the `get_original_message_from_padded_text` function directly and fudging with the padded length.